src/cmd/aem.js

```diff
@@ -16,150 +16,146 @@ import fetch from 'node-fetch';
-        },
-        handler: () => {
-            // Default handler if no valid subcommand is provided
-            console.log(chalk.red('Invalid subcommand. Use `aem login` or `aem upload`.'));
-            process.exit(1);
-        },
-    });
-}
\ No newline at end of file
+  yargs.command({
+    command: 'aem <subcommand>',
+    describe: 'Manage AEM Cloud Service interactions',
+    builder: (yargs) => {
+      return yargs
+        .command({
+          command: 'login',
+          describe: 'Login to AEM Cloud Service environment',
+          builder: (yargs) => {
+            return yargs.option('aemurl', {
+              describe: 'AEM Cloud Service URL to upload content to',
+              type: 'string',
+              demandOption: true,
+            });
+          },
+          handler: async (argv) => {
+            const credentials = await getUserCredentials();
+            console.log('Validating credentials...');
+            if (!validateLogin(argv.aemurl, credentials.username, credentials.password)) {
+              console.log(chalk.red('Invalid credentials or AEM URL.'));
+              process.exit(1);
+            }
+
+            console.log('Saving credentials...');
+            saveCredentials(argv.aemurl, credentials.username, credentials.password);
+            console.log(chalk.green('Login successful! Credentials saved securely.'));
+            process.exit(0);
+          },
+        })
+        .command({
+          command: 'upload',
+          describe: 'Upload content to AEM Cloud Service environment',
+          builder: (yargs) => yargs,
+          handler: async () => {
+            console.log('Checking for credentials...');
+            const credentials = loadCredentials();
+            if (!credentials) {
+              console.log(chalk.red('No credentials found. Run `aem login` first.'));
+              process.exit(1);
+            }
+
+            const userInputs = await getUserInputs();
+
+            console.log('Checking for files...');
+
+            if (!validateFiles(userInputs.imageMappingFile, userInputs.contentPackagePath)) {
+              console.log(chalk.green('Invalid file paths provided.'));
+              process.exit(1);
+            }
+
+            const opts = {
+              username: credentials.username,
+              password: credentials.password,
+              targetAEMUrl: credentials.url,
+              maxRetries: 3,
+            };
+
+            // Process the upload request
+            uploadImagesToAEM(opts, userInputs.imageMappingFile)
+              .then(() => uploadPackageToAEM(opts, userInputs.contentPackagePath))
+              .then(() => console.log(chalk.green('Upload completed successfully')))
+              .then(() => process.exit(0))
+              .catch((err) => {
+                console.error(chalk.red('Error during upload:', err.message));
+                process.exit(1);
+              });
```

Comment:
Since the handler is `async` can you change this two use `await` within a try/catch? Also, the `yargs` handlers shouldn't do any work but just call out to another function to do the actual work. So, in this case I could expect to see sometihng like this.

```
try {
  await runUploadContent({ ...opts, imagePath: userInputs.imageMappingFile, packagePath: userInputs.contentPackagePath});
  process.exit(0);
} catch (e) {
  console.error(chalk.red('Error during upload:', err.message));
  process.exit(1);
}
```

`runUploadContent` would then have

```
...
console.log(chalk.yellow('Uploading content to AEM...'));
await uploadImagesToAEM(...);
await uploadPackageToAEM(...);
console.log(chalk.green('Upload completed successfully'));

...
```